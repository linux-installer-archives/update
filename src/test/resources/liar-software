#!/usr/bin/env bash

#
# Copyright (C) 2017 Pierre-Yves Fourmond.

# This file is part of liar, the Linux Installer for ARchives.

# liar is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# liar is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with liar. If not, see <http://www.gnu.org/licenses/>.
#

#
# This file is used by liar-completion, which defines Bash tab auto-completion rules for the liar command.
#

declare -a software_list

software_list=(
ant
atom
bat
cheat
clion
datagrip
easy_rsa
evans
firefox
gh
glab
go
goland
graalvm
gradle
groovy
grpcui
grpcurl
hadolint
hugo
icecat
idea_community
idea_edu
idea_ultimate
jbang
jd_cmd
jdk9
jdk10
jdk11
jdk12
jdk13
jdk14
jdk15
jdk_ea16
jdk_ri7
jdk_ri8
jdk_ri9
jdk_ri10
jdk_ri11
jdk_ri12
jdk_ri13
jdk_ri14
jdk_ri15
jgo
jq
marp_cli
maven
micronaut
mongodb
node
postman
pycharm_community
pycharm_edu
pycharm_professional
robo3t
rubymine
sbt
scala
shellcheck
spring
sublimemerge
sublimetext
torbrowser
visualvm
vscode
vscodium
webstorm
xsv
youtube_dl
yq
)

ant=https://downloads.apache.org/ant/binaries/apache-ant-1.10.8-bin.zip
atom=https://github.com/atom/atom/releases/download/v1.52.0/atom-amd64.tar.gz
bat=https://github.com/sharkdp/bat/releases/download/v0.15.4/bat-v0.15.4-x86_64-unknown-linux-gnu.tar.gz
cheat=https://github.com/cheat/cheat/releases/download/4.1.0/cheat-linux-amd64.gz
clion=https://download.jetbrains.com/cpp/CLion-2020.2.3.tar.gz
datagrip=https://download-cf.jetbrains.com/datagrip/datagrip-2020.2.3.tar.gz
easy_rsa=https://github.com/OpenVPN/easy-rsa/releases/download/v3.0.7/EasyRSA-3.0.7.tgz
evans=https://github.com/ktr0731/evans/releases/download/0.9.0/evans_linux_amd64.tar.gz
firefox=https://download-installer.cdn.mozilla.net/pub/firefox/releases/82.0/linux-x86_64/fr/firefox-82.0.tar.bz2
gh=https://github.com/cli/cli/releases/download/v1.1.0/gh_1.1.0_linux_amd64.tar.gz
glab=https://github.com/profclems/glab/releases/download/v1.11.0-pre2/glab_1.11.0-pre2_Linux_x86_64.tar.gz
go=https://dl.google.com/go/go1.15.3.linux-amd64.tar.gz
goland=https://download.jetbrains.com/go/goland-2020.2.3.tar.gz
graalvm=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-20.2.0/graalvm-ce-java8-linux-amd64-20.2.0.tar.gz
gradle=https://downloads.gradle.org/distributions/gradle-6.7-bin.zip
groovy=https://dl.bintray.com/groovy/maven/apache-groovy-binary-3.0.6.zip
grpcui=https://github.com/fullstorydev/grpcui/releases/download/v1.1.0/grpcui_1.1.0_linux_x86_64.tar.gz
grpcurl=https://github.com/fullstorydev/grpcurl/releases/download/v1.7.0/grpcurl_1.7.0_linux_x86_64.tar.gz
hadolint=https://github.com/hadolint/hadolint/releases/download/v1.18.2/hadolint-Linux-x86_64
hugo=https://github.com/gohugoio/hugo/releases/download/v0.76.5/hugo_0.76.5_Linux-64bit.tar.gz
icecat=https://ftp.gnu.org/gnu/gnuzilla/60.7.0/icecat-60.7.0.en-US.gnulinux-x86_64.tar.bz2
idea_community=https://download.jetbrains.com/idea/ideaIC-2020.2.3.tar.gz
idea_edu=https://download.jetbrains.com/idea/ideaIE-2020.1.tar.gz
idea_ultimate=https://download.jetbrains.com/idea/ideaIU-2020.2.3.tar.gz
jbang=https://github.com/jbangdev/jbang/releases/download/v0.52.0/jbang-0.52.0.zip
jd_cmd=https://github.com/kwart/jd-cmd/releases/download/jd-cmd-1.1.0.Final/jd-cli-1.1.0.Final-dist.zip
jdk9=https://download.java.net/java/GA/jdk9/9.0.4/binaries/openjdk-9.0.4_linux-x64_bin.tar.gz
jdk10=https://download.java.net/java/GA/jdk10/10.0.2/19aef61b38124481863b1413dce1855f/13/openjdk-10.0.2_linux-x64_bin.tar.gz
jdk11=https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_linux-x64_bin.tar.gz
jdk12=https://download.java.net/java/GA/jdk12.0.2/e482c34c86bd4bf8b56c0b35558996b9/10/GPL/openjdk-12.0.2_linux-x64_bin.tar.gz
jdk13=https://download.java.net/java/GA/jdk13.0.2/d4173c853231432d94f001e99d882ca7/8/GPL/openjdk-13.0.2_linux-x64_bin.tar.gz
jdk14=https://download.java.net/java/GA/jdk14.0.2/205943a0976c4ed48cb16f1043c5c647/12/GPL/openjdk-14.0.2_linux-x64_bin.tar.gz
jdk15=https://download.java.net/java/GA/jdk15/779bf45e88a44cbd9ea6621d33e33db1/36/GPL/openjdk-15_linux-x64_bin.tar.gz
jdk_ea16=https://download.java.net/java/early_access/jdk16/20/GPL/openjdk-16-ea+20_linux-x64_bin.tar.gz
jdk_ri7=https://download.java.net/openjdk/jdk7u75/ri/openjdk-7u75-b13-linux-x64-18_dec_2014.tar.gz
jdk_ri8=https://download.java.net/openjdk/jdk8u40/ri/openjdk-8u40-b25-linux-x64-10_feb_2015.tar.gz
jdk_ri9=https://download.java.net/openjdk/jdk9/ri/openjdk-9+181_linux-x64_ri.zip
jdk_ri10=https://download.java.net/openjdk/jdk10/ri/openjdk-10+44_linux-x64_bin_ri.tar.gz
jdk_ri11=https://download.java.net/openjdk/jdk11/ri/openjdk-11+28_linux-x64_bin.tar.gz
jdk_ri12=https://download.java.net/openjdk/jdk12/ri/openjdk-12+32_linux-x64_bin.tar.gz
jdk_ri13=https://download.java.net/openjdk/jdk13/ri/openjdk-13+33_linux-x64_bin.tar.gz
jdk_ri14=https://download.java.net/openjdk/jdk14/ri/openjdk-14+36_linux-x64_bin.tar.gz
jdk_ri15=https://download.java.net/openjdk/jdk15/ri/openjdk-15+36_linux-x64_bin.tar.gz
jgo=https://github.com/scijava/jgo/archive/0.5.0.zip
jq=https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
marp_cli=https://github.com/marp-team/marp-cli/releases/download/v0.22.0/marp-cli-v0.22.0-linux.tar.gz
maven=https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz
micronaut=https://github.com/micronaut-projects/micronaut-starter/releases/download/v2.1.2/micronaut-cli-2.1.2.zip
# See https://www.mongodb.com/download-center/community/releases
mongodb=https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-debian10-4.7.0.tgz
node=https://nodejs.org/dist/v12.18.1/node-v12.18.1-linux-x64.tar.xz
postman=https://dl.pstmn.io/download/latest/linux64
pycharm_community=https://download.jetbrains.com/python/pycharm-community-2020.1.tar.gz
pycharm_edu=https://download.jetbrains.com/python/pycharm-edu-2020.1.tar.gz
pycharm_professional=https://download.jetbrains.com/python/pycharm-professional-2020.1.tar.gz
robo3t=https://github.com/Studio3T/robomongo/releases/download/v1.4.1/robo3t-1.4.1-linux-x86_64-122dbd9.tar.gz
rubymine=https://download.jetbrains.com/ruby/RubyMine-2020.2.3.tar.gz
sbt=https://github.com/sbt/sbt/releases/download/v1.4.0/sbt-1.4.0.zip
scala=https://downloads.lightbend.com/scala/2.13.2/scala-2.13.2.tgz
shellcheck=https://github.com/koalaman/shellcheck/releases/download/v0.7.1/shellcheck-v0.7.1.linux.x86_64.tar.xz
spring=https://repo.spring.io/release/org/springframework/boot/spring-boot-cli/2.3.3.RELEASE/spring-boot-cli-2.3.3.RELEASE-bin.zip
sublimemerge=https://download.sublimetext.com/sublime_merge_build_2025_x64.tar.xz
sublimetext=https://download.sublimetext.com/sublime_text_3_build_3211_x64.tar.bz2
torbrowser=https://www.torproject.org/dist/torbrowser/10.0.2/tor-browser-linux64-10.0.2_fr.tar.xz
visualvm=https://github.com/oracle/visualvm/releases/download/2.0.2/visualvm_202.zip
vscode=https://update.code.visualstudio.com/1.49.1/linux-x64/stable
vscodium=https://github.com/VSCodium/vscodium/releases/download/1.52.0/VSCodium-linux-x64-1.52.0.tar.gz
webstorm=https://download.jetbrains.com/webstorm/WebStorm-2020.1.tar.gz
xsv=https://github.com/BurntSushi/xsv/releases/download/0.13.0/xsv-0.13.0-x86_64-unknown-linux-musl.tar.gz
youtube_dl=https://yt-dl.org/downloads/latest/youtube-dl
yq=https://github.com/mikefarah/yq/releases/download/3.4.0/yq_linux_amd64
