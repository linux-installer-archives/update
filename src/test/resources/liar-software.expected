#!/usr/bin/env bash

#
# Copyright (C) 2017 Pierre-Yves Fourmond.

# This file is part of liar, the Linux Installer for ARchives.

# liar is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# liar is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with liar. If not, see <http://www.gnu.org/licenses/>.
#

#
# This file is used by liar-completion, which defines Bash tab auto-completion rules for the liar command.
#

declare -a last_updates
declare -a software_list

last_updates=(act atom bat cheat clion cloc datagrip dbeaver delta easy_rsa exa evans fzf firefox gh glab go goland graalvm11 graalvm17 graalvm19 gradle grails groovy grpcui grpcurl hadolint hugo hyperfine icecat idea_community idea_edu idea_ultimate jbang jd_cmd jgo jmeter jq jreleaser lazygit marp_cli maven mcs micronaut mongodb pycharm_community pycharm_edu pycharm_professional rubymine sbt scala shellcheck spring torbrowser vscodium webstorm xsv yq)

software_list=(
act
ant
atom
bat
cheat
clion
cloc
datagrip
dbeaver
delta
easy_rsa
evans
exa
firefox
fzf
gh
glab
go
goland
graalvm8
graalvm11
graalvm16
gradle
grails
groovy
grpcui
grpcurl
hadolint
helidon
hugo
hyperfine
icecat
idea_community
idea_edu
idea_ultimate
jbang
jd_cmd
jdk9
jdk10
jdk11
jdk12
jdk13
jdk14
jdk15
jdk_ea16
jdk_ea17
jdk_ri7
jdk_ri8
jdk_ri9
jdk_ri10
jdk_ri11
jdk_ri12
jdk_ri13
jdk_ri14
jdk_ri15
jgo
jmeter
jq
jreleaser
lazygit
marp_cli
maven
mcs
micronaut
mongodb
node
postman
pycharm_community
pycharm_edu
pycharm_professional
robo3t
rubymine
sbt
scala
shellcheck
spring
sublimemerge
sublimetext
torbrowser
visualvm
vscode
vscodium
webstorm
xsv
youtube_dl
yq
)

act=https://github.com/nektos/act/releases/download/v0.2.24/act_Linux_x86_64.tar.gz
ant=https://downloads.apache.org/ant/binaries/apache-ant-1.10.8-bin.zip
atom=https://github.com/atom/atom/releases/download/v1.52.1/atom-amd64.tar.gz
bat=https://github.com/sharkdp/bat/releases/download/v0.15.5/bat-v0.15.5-x86_64-unknown-linux-gnu.tar.gz
cheat=https://github.com/cheat/cheat/releases/download/4.1.1/cheat-linux-amd64.gz
clion=https://download.jetbrains.com/cpp/CLion-2020.2.4.tar.gz
cloc=https://github.com/AlDanial/cloc/releases/download/1.87/cloc-1.87.pl
datagrip=https://download-cf.jetbrains.com/datagrip/datagrip-2020.2.4.tar.gz
dbeaver=https://github.com/dbeaver/dbeaver/releases/download/21.1.4/dbeaver-ce-21.1.4-linux.gtk.x86_64.tar.gz
delta=https://github.com/dandavison/delta/releases/download/0.12.1/delta-0.12.1-x86_64-unknown-linux-gnu.tar.gz
easy_rsa=https://github.com/OpenVPN/easy-rsa/releases/download/v3.0.8/EasyRSA-3.0.8.tgz
exa=https://github.com/ogham/exa/releases/download/v0.8.1/exa-linux-x86_64-0.8.1.zip
evans=https://github.com/ktr0731/evans/releases/download/0.9.1/evans_linux_amd64.tar.gz
firefox=https://download-installer.cdn.mozilla.net/pub/firefox/releases/82.1/linux-x86_64/fr/firefox-82.1.tar.bz2
fzf=https://github.com/junegunn/fzf/releases/download/0.33.1/fzf-0.33.1-linux_arm64.tar.gz
gh=https://github.com/cli/cli/releases/download/v1.1.1/gh_1.1.1_linux_amd64.tar.gz
glab=https://github.com/profclems/glab/releases/download/v1.11.1-pre2/glab_1.11.1-pre2_Linux_x86_64.tar.gz
go=https://dl.google.com/go/go1.15.4.linux-amd64.tar.gz
goland=https://download.jetbrains.com/go/goland-2020.2.4.tar.gz
graalvm11=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-20.2.1/graalvm-ce-java11-linux-amd64-20.2.1.tar.gz
graalvm17=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.1/graalvm-ce-java17-linux-amd64-22.3.1.tar.gz
graalvm19=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.1/graalvm-ce-java19-linux-amd64-22.3.1.tar.gz
gradle=https://downloads.gradle.org/distributions/gradle-6.8-bin.zip
grails=https://github.com/grails/grails-core/releases/download/v4.0.6/grails-4.0.6.zip
groovy=https://groovy.jfrog.io/artifactory/dist-release-local/groovy-zips/apache-groovy-binary-3.0.8.zip
grpcui=https://github.com/fullstorydev/grpcui/releases/download/v1.1.1/grpcui_1.1.1_linux_x86_64.tar.gz
grpcurl=https://github.com/fullstorydev/grpcurl/releases/download/v1.7.1/grpcurl_1.7.1_linux_x86_64.tar.gz
hadolint=https://github.com/hadolint/hadolint/releases/download/v1.18.3/hadolint-Linux-x86_64
helidon=https://helidon.io/cli/latest/linux/helidon
hugo=https://github.com/gohugoio/hugo/releases/download/v0.76.6/hugo_0.76.6_Linux-64bit.tar.gz
hyperfine=https://github.com/sharkdp/hyperfine/releases/download/v1.10.1/hyperfine-v1.10.1-x86_64-unknown-linux-gnu.tar.gz
icecat=https://ftp.gnu.org/gnu/gnuzilla/60.7.1/icecat-60.7.1.en-US.gnulinux-x86_64.tar.bz2
idea_community=https://download.jetbrains.com/idea/ideaIC-2020.2.4.tar.gz
idea_edu=https://download.jetbrains.com/idea/ideaIE-2020.2.tar.gz
idea_ultimate=https://download.jetbrains.com/idea/ideaIU-2020.2.4.tar.gz
jbang=https://github.com/jbangdev/jbang/releases/download/v0.52.1/jbang-0.52.1.zip
jd_cmd=https://github.com/kwart/jd-cmd/releases/download/jd-cmd-1.1.1.Final/jd-cli-1.1.1.Final-dist.zip
jdk9=https://download.java.net/java/GA/jdk9/9.0.4/binaries/openjdk-9.0.4_linux-x64_bin.tar.gz
jdk10=https://download.java.net/java/GA/jdk10/10.0.2/19aef61b38124481863b1413dce1855f/13/openjdk-10.0.2_linux-x64_bin.tar.gz
jdk11=https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_linux-x64_bin.tar.gz
jdk12=https://download.java.net/java/GA/jdk12.0.2/e482c34c86bd4bf8b56c0b35558996b9/10/GPL/openjdk-12.0.2_linux-x64_bin.tar.gz
jdk13=https://download.java.net/java/GA/jdk13.0.2/d4173c853231432d94f001e99d882ca7/8/GPL/openjdk-13.0.2_linux-x64_bin.tar.gz
jdk14=https://download.java.net/java/GA/jdk14.0.2/205943a0976c4ed48cb16f1043c5c647/12/GPL/openjdk-14.0.2_linux-x64_bin.tar.gz
jdk15=https://download.java.net/java/GA/jdk15/779bf45e88a44cbd9ea6621d33e33db1/36/GPL/openjdk-15_linux-x64_bin.tar.gz
jdk_ea16=https://download.java.net/java/early_access/jdk16/20/GPL/openjdk-16-ea+20_linux-x64_bin.tar.gz
jdk_ea17=https://download.java.net/java/early_access/jdk17/2/GPL/openjdk-17-ea+2_linux-x64_bin.tar.gz
jdk_ri7=https://download.java.net/openjdk/jdk7u75/ri/openjdk-7u75-b13-linux-x64-18_dec_2014.tar.gz
jdk_ri8=https://download.java.net/openjdk/jdk8u40/ri/openjdk-8u40-b25-linux-x64-10_feb_2015.tar.gz
jdk_ri9=https://download.java.net/openjdk/jdk9/ri/openjdk-9+181_linux-x64_ri.zip
jdk_ri10=https://download.java.net/openjdk/jdk10/ri/openjdk-10+44_linux-x64_bin_ri.tar.gz
jdk_ri11=https://download.java.net/openjdk/jdk11/ri/openjdk-11+28_linux-x64_bin.tar.gz
jdk_ri12=https://download.java.net/openjdk/jdk12/ri/openjdk-12+32_linux-x64_bin.tar.gz
jdk_ri13=https://download.java.net/openjdk/jdk13/ri/openjdk-13+33_linux-x64_bin.tar.gz
jdk_ri14=https://download.java.net/openjdk/jdk14/ri/openjdk-14+36_linux-x64_bin.tar.gz
jdk_ri15=https://download.java.net/openjdk/jdk15/ri/openjdk-15+36_linux-x64_bin.tar.gz
jgo=https://github.com/scijava/jgo/archive/0.5.1.zip
jmeter=https://mirroir.wptheme.fr/apache/jmeter/binaries/apache-jmeter-5.4.4.zip
jq=https://github.com/stedolan/jq/releases/download/jq-1.7/jq-linux64
jreleaser=https://github.com/jreleaser/jreleaser/releases/download/v0.1.1/jreleaser-0.1.1.zip
lazygit=https://github.com/jesseduffield/lazygit/releases/download/v0.35/lazygit_0.35_Linux_x86_64.tar.gz
marp_cli=https://github.com/marp-team/marp-cli/releases/download/v0.22.1/marp-cli-v0.22.1-linux.tar.gz
maven=https://downloads.apache.org/maven/maven-3/3.6.4/binaries/apache-maven-3.6.4-bin.tar.gz
mcs=https://github.com/mthmulders/mcs/releases/download/v0.1.14/mcs-0.1.14-linux-x86_64.tar.gz
micronaut=https://github.com/micronaut-projects/micronaut-starter/releases/download/v2.1.3/micronaut-cli-2.1.3.zip
# See https://www.mongodb.com/download-center/community/releases
mongodb=https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-debian10-4.7.1.tgz
node=https://nodejs.org/dist/v12.18.1/node-v12.18.1-linux-x64.tar.xz
postman=https://dl.pstmn.io/download/latest/linux64
pycharm_community=https://download.jetbrains.com/python/pycharm-community-2020.2.tar.gz
pycharm_edu=https://download.jetbrains.com/python/pycharm-edu-2020.2.tar.gz
pycharm_professional=https://download.jetbrains.com/python/pycharm-professional-2020.2.tar.gz
robo3t=https://github.com/Studio3T/robomongo/releases/download/v1.4.1/robo3t-1.4.1-linux-x86_64-122dbd9.tar.gz
rubymine=https://download.jetbrains.com/ruby/RubyMine-2020.2.4.tar.gz
sbt=https://github.com/sbt/sbt/releases/download/v1.4.1/sbt-1.4.1.zip
scala=https://downloads.lightbend.com/scala/2.13.3/scala-2.13.3.tgz
shellcheck=https://github.com/koalaman/shellcheck/releases/download/v0.7.2/shellcheck-v0.7.2.linux.x86_64.tar.xz
spring=https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-cli/3.0.7/spring-boot-cli-3.0.7-bin.zip
sublimemerge=https://download.sublimetext.com/sublime_merge_build_2025_x64.tar.xz
sublimetext=https://download.sublimetext.com/sublime_text_3_build_3211_x64.tar.bz2
sublimetext4=https://download.sublimetext.com/sublime_text_build_4107_x64.tar.xz
torbrowser=https://www.torproject.org/dist/torbrowser/10.0.3/tor-browser-linux64-10.0.3_fr.tar.xz
visualvm=https://github.com/oracle/visualvm/releases/download/2.0.2/visualvm_202.zip
vscode=https://update.code.visualstudio.com/1.49.1/linux-x64/stable
vscodium=https://github.com/VSCodium/vscodium/releases/download/1.52.1/VSCodium-linux-x64-1.52.1.tar.gz
webstorm=https://download.jetbrains.com/webstorm/WebStorm-2020.2.tar.gz
xsv=https://github.com/BurntSushi/xsv/releases/download/0.13.1/xsv-0.13.1-x86_64-unknown-linux-musl.tar.gz
youtube_dl=https://yt-dl.org/downloads/latest/youtube-dl
yq=https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
